plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeAnalyzeProperties.java

```diff
@@ -76,6 +83,11 @@ public static Optional<Instant> getFilesModifiedAfterProperty(Map<String, Object
         return Optional.ofNullable((Instant) properties.get(FILES_MODIFIED_AFTER));
     }
 
+    public static Optional<Boolean> getFullRefresh(Map<String, Object> properties)
+    {
+        return Optional.ofNullable((Boolean) properties.get(FULL_REFRESH));
```

Comment:
I suppose there's no chance to get null because we set the default value (false) in the above `booleanProperty`. I would change to non-Optional and throw an exception in case of null as it indicates something is wrong.